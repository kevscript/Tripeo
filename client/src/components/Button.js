import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

const ButtonContainer = styled.button`
  font-family: 'Source Sans Pro', 'Roboto', sans-serif;
  border: 0;
  outline: 0;
  cursor: pointer;
  padding: ${props => props.big ? '15px 45px' : '15px 25px'};
  font-size: ${props => props.big ? '26px' : '15px'};
  border-radius: 50px;
  background: ${props => props.theme.colors.primary};
  color: ${props => props.theme.text.light};
  min-width: 100px;
  transition: all 0.2s ease-in-out;
  font-weight: 600;

  &:disabled {
    cursor: not-allowed;
    background: rgba(0,0,0,0.1);
  }

  &:hover:not(:disabled) {
    background: ${props => props.theme.colors.primaryDark};
    transform: translateY(-5px);
  }
`

ButtonContainer.displayName = 'ButtonContainer'

const Button = ({
  children,
  big = false,
  handleClick,
  disabled = false
}) => {
  return (
    <ButtonContainer
      big={big}
      disabled={disabled}
      onClick={handleClick}
      data-test='Button'
    >
      {children ? children : 'Button'}
    </ButtonContainer>
  )
}

Button.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.node
  ]),
  big: PropTypes.bool,
  disabled: PropTypes.bool,
  handleClick: PropTypes.func
}

export default Button