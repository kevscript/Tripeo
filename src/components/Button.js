import React from 'react'
import styled from 'styled-components'

const ButtonContainer = styled.div`
  cursor: pointer;
  margin: 2rem 0;
  padding: 15px 45px;
  font-size: 2.6rem;
  border-radius: 50px;
  border: 1px solid rgba(0,0,0,0.1);
  background: lightcoral;
  color: papayawhip;
`

ButtonContainer.displayName = 'ButtonContainer'

const Button = ({ children }) => {
  return (
    <ButtonContainer>
      {children}
    </ButtonContainer>
  )
}

export default Button