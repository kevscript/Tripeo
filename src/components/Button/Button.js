import React from 'react'
import styled from 'styled-components'

const ButtonContainer = styled.button`
  cursor: pointer;
`

ButtonContainer.displayName = 'ButtonContainer'

const Button = ({text}) => {
  return (
    <ButtonContainer>
      {text}
    </ButtonContainer>
  )
}

export default Button