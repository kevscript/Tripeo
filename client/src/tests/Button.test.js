import React from 'react'
import { shallow } from 'enzyme'
import Button from '../components/Button'


describe('Button Component', () => {

  describe('when it has props', () => {

    const props = {
      children: 'Hello',
      big: true,
      disabled: false,
      handleClick: jest.fn()
    }

    let component
    let wrapper

    beforeEach(() => {
      component = shallow(<Button {...props} />)
      wrapper = component.find('[data-test="Button"]')
    })

    test('should exist', () => {
      expect(wrapper.length).toBe(1)
    })

    test('should renders correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })

    test('should render the correct text', () => {
      expect(wrapper.text()).toBe(props.children)
    })

    test('should render with correct props', () => {
      expect(wrapper.prop('big')).toBe(true)
      expect(wrapper.prop('disabled')).toBe(false)
    })

    test('should call the right click handler', () => {
      wrapper.simulate('click')
      expect(props.handleClick).toHaveBeenCalled()
    })
  })


  describe('when it has NO props', () => {

    let component
    let wrapper

    beforeEach(() => {
      component = shallow(<Button />)
      wrapper = component.find('[data-test="Button"]')
    })

    test('should render with correct default text', () => {
      expect(wrapper.text()).toBe('Button')
    })

    test('should renders correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })


    test('should assign the default props', () => {
      expect(wrapper.prop('big')).toBe(false)
      expect(wrapper.prop('disabled')).toBe(false)
    })
  })
})