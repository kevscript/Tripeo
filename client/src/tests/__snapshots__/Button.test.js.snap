// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component when it has NO props should renders correctly 1`] = `ShallowWrapper {}`;

exports[`Button Component when it has props should renders correctly 1`] = `ShallowWrapper {}`;
